httpserver/handlers.go

```diff
@@ -83,19 +87,96 @@ func (server *Server) verify(ctx context.Context, w http.ResponseWriter, r *http
 				Key:   subject,
 				Value: fromVerifyResult(result),
 			})
+			logrus.Debugf("verification: execution time for image %s: %dms", subject, time.Since(routineStartTime).Milliseconds())
 		}(utils.SanitizeString(subject))
 	}
 	wg.Wait()
+	logrus.Debugf("verification: execution time for request: %dms", time.Since(startTime).Milliseconds())
+	return sendResponse(&results, "", w, http.StatusOK, false)
+}
+
+func (server *Server) mutate(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	startTime := time.Now()
+	sanitizedMethod := utils.SanitizeString(r.Method)
+	sanitizedURL := utils.SanitizeURL(*r.URL)
+	logrus.Infof("start request %s %s", sanitizedMethod, sanitizedURL)
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("unable to read request body: %v", err)
+	}
+
+	// parse request body
+	var providerRequest externaldata.ProviderRequest
+	err = json.Unmarshal(body, &providerRequest)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal request body: %v", err)
+	}
 
-	return sendResponse(&results, "", w, http.StatusOK)
+	results := make([]externaldata.Item, 0)
+	wg := sync.WaitGroup{}
+	mu := sync.RWMutex{}
```

Comment:
Why ReadWrite mutex? I see only the write lock being used.

Reply:
switched to regular `mutex`